src/resolve/resolved-dns-query.c

```diff
@@ -592,6 +592,11 @@ int dns_query_new(
                 if (question_utf8 || question_idna)
                         return -EINVAL;
 
+                _unused_ _cleanup_(dns_question_unrefp) DnsQuestion *filtered_question_bypass = NULL;
+                r = manager_validate_and_mangle_question(m, &question_bypass->question, &filtered_question_bypass);
```

Comment:
This will leak previously assigned `question_bypass->question`, and triggers use-after-free of `filtered_question_bypass`.